x-pack/filebeat/input/http_endpoint/validate.go

```diff
@@ -62,17 +62,15 @@ func (v *apiValidator) validateRequest(r *http.Request) (status int, err error)
 	}
 
 	if v.hmacHeader != "" && v.hmacKey != "" && v.hmacType != "" {
-		// Read HMAC signature from HTTP header.
-		hmacHeaderValue := r.Header.Get(v.hmacHeader)
-		if v.hmacHeader == "" {
```

Comment:
This was the problematic line. It was checking `v.hmacHeader` instead of `hmacHeaderValue`.